src/dcqc/target.py

```diff
@@ -67,12 +67,13 @@ def get_file_type(self) -> FileType:
 
     @wraps(File.stage)
     def stage(
-        self,
-        destination: Optional[Path] = None,
-        overwrite: bool = False,
-    ) -> Iterator[Path]:
+        self, destination: Optional[Path] = None, overwrite: bool = False
+    ) -> list[Path]:
+        paths = list()
         for file in self.files:
-            yield file.stage(destination, overwrite)
```

Comment:
Why the switch from generator?

Reply:
Because calling `paths = target.stage()` wouldn't stage the files. Instead, it would return an iterator that you would then need to iterate though (_e.g._ using `list(paths)`) to actually stage the files. I don't think this is intuitive. In any case, there is no real benefit to a generator here since it's not as if large amounts of data are being processed in-memory. Also, most targets only have a single file, so it makes even less sense most of the time.

Reply:
Makes sense. Thanks!